### PR TITLE
Support setting ancestor IDs

### PR DESCRIPTION
**Pull request by rockapot**
- Ancestor ID setter is `setAncestorIds`
- Rename `getLoggingSessionInfo` to `getCurrentOrPendingAncestorIds`
- Rename collection view methods to `collectionViewDidMount` and `collectionViewWillUnmount` to bring them in line with React Native terminology
- Convert BuildProject's AppDelegate from Objective C to Swift
- Expose `promotedViewabilityConfig` from `useImpressionLogger`

### CODE REVIEW

File: ios/PromotedMetricsModule.swift [hunk 1]
```diff
@@ -265,32 +265,47 @@ public extension PromotedMetricsModule {
...
+                                   rejecter: RCTPromiseRejectBlock) {
     guard let metricsLogger = metricsLogger else {
       resolver([:])
       return
     }
     let sessionInfo = [
-      "logUserId": metricsLogger.logUserID,
-      "sessionId": metricsLogger.sessionID,
-      "viewId": metricsLogger.viewID,
+      "logUserId": metricsLogger.currentOrPendingLogUserID,
```
Comment (danbosnichill) [inline ios/PromotedMetricsModule.swift:291]: not a blocker - this is the type of scoping that I was expecting for these fields.  When I think about logUserId - I think about 'currentOrPending' version of it.  If this is the default, we can rename the normal field ('logUserID') to this behavior.

**Review by danbosnichill — APPROVED** [review]

File: BuildProject/App.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@
  */
 
 import PromotedMetrics, { ActionType } from '@promotedai/react-native-metrics';
+import type { AncestorIds } from '@promotedai/react-native-metrics';
```
Comment (jinyius) [inline BuildProject/App.tsx:9]: is there an autoformatter for rn?

File: ios/PromotedMetricsModule.m [hunk 3]
```diff
@@ -82,16 +82,18 @@ @interface RCT_EXTERN_REMAP_MODULE(PromotedMetrics, PromotedMetricsModule, NSObj
                   useCase:(NSInteger)useCase)
 
 #pragma mark - ImpressionLogger
-RCT_EXTERN_METHOD(collectionViewDidLoad:(NSString *)collectionViewName)
+RCT_EXTERN_METHOD(collectionViewDidMount:(NSString *)collectionViewName)
```
Comment (jinyius) [inline ios/PromotedMetricsModule.m:85]: have you looked into using some lexer or something to determine method name consistency at the code locations that need it?

**Review by jinyius — APPROVED** [review]

File: ios/PromotedMetricsModule.swift (hunk 1 above)
Comment (rockapot) [reply ios/PromotedMetricsModule.swift:291]: What would we call the other version of the field?

File: BuildProject/App.tsx (hunk 2 above)
Comment (rockapot) [reply BuildProject/App.tsx:9]: Yeah, I can add a lint target to the package.json.

File: ios/PromotedMetricsModule.m (hunk 3 above)
Comment (rockapot) [reply ios/PromotedMetricsModule.m:85]: The tests should cover this. I can think of a solution that improves the macro, maybe I'll suggest it to the RN community.

**Pull request merged**